bitwarden_license/src/CommCore/Services/ProviderService.cs

```diff
@@ -120,29 +121,36 @@ public async Task UpdateAsync(Provider provider, bool updateBilling = false)
             await _providerRepository.ReplaceAsync(provider);
         }
 
-        public async Task<List<ProviderUser>> InviteUserAsync(Guid providerId, Guid invitingUserId,
-            ProviderUserInvite invite)
+        public async Task<List<ProviderUser>> InviteUserAsync(ProviderUserInvite<string> invite)
         {
-            var provider = await _providerRepository.GetByIdAsync(providerId);
-            if (provider == null || invite?.Emails == null || !invite.Emails.Any())
+            if (!_currentContext.ProviderManageUsers(invite.ProviderId))
+            {
+                throw new BadRequestException("Invalid permissions.");
```

Comment:
This would be the first place we throw a `BadRequestException` with the message `Invalid permissions.` I don't think it's a bad message, but it's not in line with what we're currently doing.

I believe the more typical exception is actually a `NotFoundException`, which I don't like. Maybe we should ping the team and get opinions on error to throw on lacking permissions?

Reply:
I ended up running with an InvalidOperationException, since that seems to fit the bill for what is actually happening here based on [MS docs](https://docs.microsoft.com/en-us/dotnet/api/system.invalidoperationexception?view=net-5.0#throwing-an-invalidoperationexception-exception).

I also changed an ApplicationException I saw to an ArguementException since papa Nadella says [not to throw or derive ApplicationException](https://docs.microsoft.com/en-us/dotnet/api/system.applicationexception?view=net-5.0).

Reply:
Saving real discussion for this for PIQ stuff, but I'm beginning to think we should set up a formal process for addressing tech debt and cleaning up after design and code style changes.